pkg/lib/accountmanagement/service_identity.go

```diff
@@ -0,0 +1,1333 @@
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &DeleteIdentityPhoneOutput{IdentityInfo: info}, nil
+}
+
+type AddPasskeyInput struct {
+	CreationResponse *protocol.CredentialCreationResponse
+}
+
+type AddPasskeyOutput struct {
+	IdentityInfo *identity.Info
+}
+
+func (s *Service) AddPasskey(resolvedSession session.ResolvedSession, input *AddPasskeyInput) (*AddPasskeyOutput, error) {
+	userID := resolvedSession.GetAuthenticationInfo().UserID
+	creationResponse := input.CreationResponse
+	creationResponseBytes, err := json.Marshal(creationResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	authenticatorSpec := &authenticator.Spec{
+		UserID: userID,
+		Kind:   authenticator.KindPrimary,
+		Type:   model.AuthenticatorTypePasskey,
+		Passkey: &authenticator.PasskeySpec{
+			AttestationResponse: creationResponseBytes,
+		},
+	}
+
+	authenticatorID := uuid.New()
+	authenticatorInfo, err := s.Authenticators.NewWithAuthenticatorID(authenticatorID, authenticatorSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	identitySpec := &identity.Spec{
+		Type: model.IdentityTypePasskey,
+		Passkey: &identity.PasskeySpec{
+			AttestationResponse: creationResponseBytes,
+		},
+	}
+
+	var identityInfo *identity.Info
+	err = s.Database.WithTx(func() error {
+		identityInfo, err = s.prepareNewIdentity(userID, identitySpec)
+		if err != nil {
+			return err
+		}
+
+		err = s.createIdentity(identityInfo)
+		if err != nil {
+			return err
+		}
+
+		err = s.dispatchIdentityCreatedEvent(identityInfo)
+		if err != nil {
+			return err
+		}
+
+		err = s.Authenticators.Create(authenticatorInfo, false)
+		if err != nil {
+			return err
+		}
+		err = s.PasskeyService.ConsumeAttestationResponse(creationResponseBytes)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &AddPasskeyOutput{IdentityInfo: identityInfo}, nil
+}
+
+type DeletePasskeyInput struct {
+	IdentityID string
+}
+
+type DeletePasskeyOutput struct {
+	IdentityInfo *identity.Info
+}
+
+func (s *Service) DeletePasskey(resolvedSession session.ResolvedSession, input *DeletePasskeyInput) (*DeletePasskeyOutput, error) {
+	userID := resolvedSession.GetAuthenticationInfo().UserID
+	identityID := input.IdentityID
+
+	var info *identity.Info
+	err := s.Database.WithTx(func() (err error) {
+		info, err = s.prepareDeleteIdentity(userID, identityID)
+		if err != nil {
+			return err
+		}
+
+		err = s.deleteIdentity(info)
+		if err != nil {
+			return err
+		}
+
+		err = s.dispatchIdentityDeletedEvent(info)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &DeletePasskeyOutput{IdentityInfo: info}, nil
+}
+
+type AddIdentityBiometricInput struct {
+	JWTToken string
+}
+
+type AddIdentityBiometricOutput struct {
+	IdentityInfo *identity.Info
+}
+
+func (s *Service) AddIdentityBiometric(resolvedSession session.ResolvedSession, input *AddIdentityBiometricInput) (*AddIdentityBiometricOutput, error) {
```

Comment:
I do not think we support adding biometric identity in the settings page. Remove code related to adding biometric identity.

Reply:
oh right